package/MDAnalysis/analysis/Aggregation_size.py

```diff
@@ -0,0 +1,370 @@
+        connected molecules in a cluster.
+        k is a connected molecule to the main molecule in the first level.
+        self._frame_index is a time frame.
+        counter is a step.
+        m provides a suitable off_set to match the number of molecules.
+        Limit is the cut_off provided by the user.
+        """
+        bool_sector = self.d[k] < cut_off
+        Coordinates = self.d[k][bool_sector]
+        # This "if" will check if there is any uninvestigated molecule left
+        if len(Coordinates) != 0:
+            index1 = np.where(bool_sector)
+            listOfCoordinates = list(zip(index1[0], index1[1]))
+            connect = ([j[0] for j in listOfCoordinates]
+                       + connected[m] + 1)
+            Connected_NLevel = np.array(connect)
+        else:
+            Connected_NLevel = []
+        return Connected_NLevel
+
+    def _single_frame(self):
+        """
+        1- Import the needed module
+        2- Calculate the distance between molecules based on the intended
+        criteria
+        3- Find the aggregates by checking only one time each using the
+        "aggregate" subfunction (to be efficient)
+        4- Check and merge two aggregates as one aggregate if the distance
+        between any element of them is less than cut_off
+        5- Eliminate monomers (optional)
+        6- measure average gyration radius (optional)
+        7- measure the number of aggregates with different size
+        """
+        aggregate = [0] * self._molnum
+        if self._type == "Closest":
+            self.closest_atom(self.d)
+        elif self._type == "COM":
+            self.center_of_mass(self.d)
+        elif self._type == "Specific":
+            self.specific_atoms(self.d)
+        else:
+            raise ValueError("enter valid Type")
+        # P_Level collact all the investigated nodes in past levels
+        P_Level = []
+        # Group will collect the molecule first investigating molecule
+        # len(Group) will be the number of clusters
+        Group = []
+        # this for will go through all molecules to
+        # check their connection with others
+        for counter in range(1, self._molnum):
+            # aggergate parameter include the molecules in each cluster
+            aggregate[counter-1] = [counter]
+            # this "if" will restrict double calculation of the same molecule
+            # to increase efficincy of code
+            if counter not in P_Level:
+                # define the ones that are attached
+                # to a molecule in first level
+                bool_sector = self.d[counter] < self._cut_off
+                Coordinates = self.d[counter][bool_sector]
+                index1 = np.where(bool_sector)
+                # list the connected molecules to a molecule in first level
+                listOfCoordinates = list(zip(index1[0], index1[1]))
+                # match array number with molecule number: 2-3[0] => 2-3 [3]
```

Comment:
I don't understand this comment; what does "2-3" mean? Explicit comments are always better than short, but possibly confusing, ones. It's also often easier to use the variable names in the code, e.g. `p_level-d_level[counter]-> p_level-d_level[connected]` (just an example, as I do not know what you're referring to here)

Reply:
This comment we assume that we are studying the connected molecule with molecule 2, and we assume molecule 2 is connected to molecule 3. In this case, the first element of "self.d[1]", which has 0 index, will show up in "i[0] for i in listOfcoordinate])". But we would like to save it as 3 for further calculation. So, we are doing the next line code to adjust the number of the index (e.g. 0) with the number of the connected molecule (e.g. 3) to the main molecule (e.g. 2), if that makes sense. I will try to make it more clear in the code as well.